apps/dashboard/src/components/header-navigation/edit-bridge-url-button.tsx

```diff
@@ -0,0 +1,123 @@
+  const { currentEnvironment, setBridgeUrl } = useEnvironment();
+  const { status, bridgeURL: envBridgeUrl } = useBridgeHealthCheck();
+  const { validateBridgeUrl, isLoading: isValidatingBridgeUrl } = useValidateBridgeUrl();
+  const { updateBridgeUrl, isLoading: isUpdatingBridgeUrl } = useUpdateBridgeUrl();
+
+  useLayoutEffect(() => {
+    reset({ bridgeUrl: envBridgeUrl });
+  }, [reset, envBridgeUrl]);
+
+  const onSubmit: SubmitHandler<FormFields> = async ({ bridgeUrl }) => {
+    const { isValid } = await validateBridgeUrl(bridgeUrl);
+    if (isValid) {
+      await updateBridgeUrl({ url: bridgeUrl, environmentId: currentEnvironment?._id ?? '' });
+      setBridgeUrl(bridgeUrl);
+    } else {
+      setError('bridgeUrl', { message: 'The provided URL is not the Novu Endpoint URL' });
+    }
+  };
+
+  return (
+    <Popover
+      open={isPopoverOpen}
+      onOpenChange={(newIsOpen) => {
+        setIsPopoverOpen(newIsOpen);
+        if (!newIsOpen && isDirty) {
+          reset({ bridgeUrl: envBridgeUrl });
+        }
+      }}
+    >
+      <PopoverTrigger asChild>
+        <button className="text-foreground-600 flex h-6 items-center gap-2 rounded-md border border-neutral-200 text-xs leading-4 hover:bg-neutral-50 focus:bg-neutral-50">
+          <div className="flex items-center gap-2 px-1.5 py-1">
+            <span
+              className={cn(
+                'relative size-1.5 animate-[pulse-shadow_1s_ease-in-out_infinite] rounded-full',
+                status === ConnectionStatus.DISCONNECTED ? 'bg-destructive' : 'bg-success'
+              )}
+              style={
+                {
+                  '--pulse-color': status === ConnectionStatus.DISCONNECTED ? 'var(--destructive)' : 'var(--success)',
+                } as React.CSSProperties
+              }
+            />
+            <span>Local Studio</span>
+          </div>
+          <span className="border-l border-neutral-200 p-1.5">
+            <RiPencilFill className="size-[12px]" />
+          </span>
+        </button>
+      </PopoverTrigger>
+      <PopoverPortal>
+        <PopoverContent className="w-[362px] p-0" side="bottom" align="end">
+          <form onSubmit={handleSubmit(onSubmit)}>
+            <div className="p-5">
+              <div className="flex flex-col gap-1">
+                <InputGroup>
+                  <InputLabel htmlFor="bridgeUrl">Bridge Endpoint URL</InputLabel>
+                  <InputField variant="xs" state={errors.bridgeUrl?.message ? 'error' : 'default'}>
+                    <RiLinkM className="size-5 min-w-5" />
+                    <Input id="bridgeUrl" {...register('bridgeUrl')} />
+                  </InputField>
+                  <InputHint state={errors.bridgeUrl?.message ? 'error' : 'default'}>
+                    {errors.bridgeUrl?.message ?? 'Full path URL (e.g., https://your.api.com/api/novu)'}
+                  </InputHint>
+                </InputGroup>
```

Comment:
The newly added Input field will allow us to build any input variant from the designs. Currently added the regular input with: label, hint, and disabled and error states. Names TBD.